cherrypy/_cpreqbody.py

```diff
@@ -832,8 +876,12 @@ def read(self, size=None, fp_out=None):
             # Check lengths.
             self.bytes_read += datalen
             if self.maxbytes and self.bytes_read > self.maxbytes:
-                raise cherrypy.HTTPError(413)
-
+                if self.maxbytes_extend:
```

Comment:
This block of code duplicates code above. It must be moved to separate private/protected method